### ISSUE TITLE
execute sql with  impossible where

### ISSUE THREAD

**Issue by wsabc01** [pingcap/tidb#4977]

**Comment by wsabc01** [comment]
There are some sql with impossible where of zabbix ,for example 'where xxx and 1=0'. When use mysql, it will return result quickly,but for TiDB,it will calculate the previous condition ,this costs a long time. For zabbix which used TiDB ,this sql would get a return of time out.

####Explain Infomation：
##mysql：
```
explain SELECT t.triggerid FROM triggers t,functions f,items i,hosts h WHERE t.triggerid=f.triggerid AND f.itemid=i.itemid AND i.hostid=h.hostid AND (i.status=1 OR h.status=1 OR t.status=1) AND 1=0 ;
+----+-------------+-------+------------+------+---------------+------+---------+------+------+----------+------------------+
| id | select_type | table | partitions | type | possible_keys | key  | key_len | ref  | rows | filtered | Extra            |
+----+-------------+-------+------------+------+---------------+------+---------+------+------+----------+------------------+
|  1 | SIMPLE      | NULL  | NULL       | NULL | NULL          | NULL | NULL    | NULL | NULL |     NULL | Impossible WHERE |
+----+-------------+-------+------------+------+---------------+------+---------+------+------+----------+------------------+
```

##TiDB
```
explain SELECT t.triggerid FROM triggers t,functions f,items i,hosts h WHERE t.triggerid=f.triggerid AND f.itemid=i.itemid AND i.hostid=h.hostid AND (i.status=1 OR h.status=1 OR t.status=1) AND 1=0 ;
+------------------+------------------+---------------------------------+------+----------------------------------------------+--------+
| id               | parents          | children                        | task | operator info                                | count  |
+------------------+------------------+---------------------------------+------+----------------------------------------------+--------+
| TableScan_19     |                  |                                 | cop  | table:t, range:(-inf,+inf), keep order:false |    364 |
| TableReader_20   | HashRightJoin_18 |                                 | root | data:TableScan_19                            |    364 |
| TableScan_21     |                  |                                 | cop  | table:f, range:(-inf,+inf), keep order:false |   1164 |
| TableReader_22   | HashRightJoin_18 |                                 | root | data:TableScan_21                            |   1164 |
| HashRightJoin_18 | HashLeftJoin_15  | TableReader_20,TableReader_22   | root | inner join, small:TableReader_20             | 423696 |
| TableScan_23     |                  |                                 | cop  | table:i, range:(-inf,+inf), keep order:false |      0 |
| TableReader_24   | HashLeftJoin_15  |                                 | root | data:TableScan_23                            |      0 |
| HashLeftJoin_15  | HashLeftJoin_13  | HashRightJoin_18,TableReader_24 | root | inner join, small:TableReader_24             |      0 |
| TableDual_25     | HashLeftJoin_13  |                                 | root | rows:0                                       |      0 |
| HashLeftJoin_13  | Projection_9     | HashLeftJoin_15,TableDual_25    | root | inner join, small:TableDual_25               |      0 |
| Projection_9     |                  | HashLeftJoin_13                 | root | t.triggerid                                  |      0 |
+------------------+------------------+---------------------------------+------+----------------------------------------------+--------+
```

## Version ：
```
Release Version: v1.0.0-73-g55403c8
Git Commit Hash: 55403c8469fc71d3cda303338ccd5bde31357027
Git Branch: master
UTC Build Time: 2017-10-30 04:47:28
```

**Comment by shenli** [comment]
@wsabc01 Thanks for your feedback!
@hanfei1991 @winoros PTAL

**Comment by shenli** [comment]
@wsabc01 This is fixed in #4980 . Please try the latest master. 
Thanks!
